src/components/progress/linear-progress.ts

```diff
@@ -0,0 +1,100 @@
+import { html, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import { asPercent, partNameMap } from '../common/util';
+import { IgcProgressBaseComponent } from './base';
+import { styles } from './themes/linear/linear.progress.base.css';
+import { styles as bootstrap } from './themes/linear/linear.progress.bootstrap.css';
+import { styles as fluent } from './themes/linear/linear.progress.fluent.css';
+import { styles as indigo } from './themes/linear/linear.progress.indigo.css';
+import { themes } from '../../theming';
+import { styleMap } from 'lit/directives/style-map.js';
+
+/**
+ * A linear progress indicator used to express unspecified wait time or display
+ * the length of a process.
+ *
+ * @element igc-linear-progress
+ *
+ * @slot - The text area container.
+ *
+ * @csspart track
+ * @csspart fill
+ * @csspart striped
+ * @csspart label
+ * @csspart value
+ * @csspart indeterminate
+ * @csspart primary
+ * @csspart danger
+ * @csspart warning
+ * @csspart info
+ * @csspart success
+ * @csspart top
+ * @csspart top-start
+ * @csspart top-end
+ * @csspart bottom
+ * @csspart bottom-start
+ * @csspart bottom-end
```

Comment:
Do we apply these parts?

Reply:
Actually not anymore. I'll remove them in a second.